modules/server/src/network/aggregations/index.ts

```diff
@@ -44,21 +44,46 @@ export const resolveToNetworkAggregation = (
  * Takes an array of the same aggregation type and computes the singular type
  * eg. NumericAggregation => NetworkNumericAggregation
  *
+ * Note for operations on Buckets -  size can be large
+ *
  * @param aggregations
  * @returns
  */
 export const resolveAggregation = (aggregations: Aggregations[]): NetworkAggregation => {
 	const emptyAggregation: NetworkAggregation = { bucket_count: 0, buckets: [] };
 
 	const resolvedAggregation = aggregations.reduce((resolvedAggregation, agg) => {
-		const computedBucketCount = resolvedAggregation.bucket_count + agg.bucket_count;
-		const computedBuckets = agg.buckets.map(({ key, doc_count }) => {
-			// potentially expensive "find" if array of buckets is very large
-			const bucket = resolvedAggregation.buckets.find((bucket) => bucket.key === key);
-			return { key, doc_count: (bucket?.doc_count || 0) + doc_count };
+		const bucketCountAccumulator = resolvedAggregation.bucket_count + agg.bucket_count;
+
+		/*
+		 * Unable to use lookup key eg. buckets[key]
+		 * "buckets": [
+		 *  {
+		 *    "doc_count": 140,
+		 *    "key": "Dog"
+		 *   },
+		 */
+		const computedBuckets = resolvedAggregation.buckets;
+
+		agg.buckets.forEach((bucket) => {
+			const { key, doc_count } = bucket;
+			const existingBucketIndex = computedBuckets.findIndex((bucket) => bucket.key === key);
+			if (existingBucketIndex !== -1) {
+				const existingBucket = computedBuckets[existingBucketIndex];
+				if (existingBucket) {
+					// update existing bucket
+					computedBuckets[existingBucketIndex] = {
+						...existingBucket,
+						doc_count: existingBucket.doc_count + doc_count,
+					};
+				}
+			} else {
+				computedBuckets.push(bucket);
+			}
```

Comment:
Size of buckets can be large
Using `forEach` to either update "found bucket" or add to resolved buckets

Avoids using functional operations on the entire "resolved buckets" and then having a large array of Buckets and needing to flatten and filter on same Bucket `key`

No way around searching buckets for existing bucket unfortunately as the data structure is an array and a key lookup is not an option.

Reply:
the lookup has always been a pain point... perhaps we should make a ticket for future improvement to make the buckets a dictionary available internally in the Arranger server for this and other lookups, even if it's ultimately turned in the UI as an array for display purposes.

Reply:
https://github.com/overture-stack/arranger/issues/886